### PR TITLE
disable text-input, switch and button in entity-information if not qu…

### PR DESCRIPTION
**Pull request by kiennguyen2403**
…alified

### CODE REVIEW

File: lib/entity/config/config_list_item.dart [hunk 1]
```diff
@@ -89,17 +91,23 @@ class ConfigListItem extends ConsumerWidget {
...
-                        ? FirebaseFirestore.instance
-                            .runTransaction((Transaction myTransaction) async {
-                            myTransaction.delete(FirebaseFirestore.instance
-                                .collection('entity')
-                                .doc(entityId)
-                                .collection(configType)
-                                .doc(configDoc.id));
-                          })
-                        : addEntity(context, ref, configDoc);
+                  onChanged: (value){
```
Comment (kovarny) [inline lib/entity/config/config_list_item.dart:94]: pls change to assing null to onChanged to disable it.

**Review by kovarny — CHANGES_REQUESTED** [review]

File: lib/entity/config/config_list_item.dart [hunk 2]
```diff
@@ -5,15 +5,23 @@ import 'package:seeder/controls/doc_field_range_slider.dart';
 import 'package:seeder/controls/doc_field_slider.dart';
 import 'package:seeder/controls/group.dart';
 import 'package:seeder/providers/firestore.dart';
+import 'package:firebase_auth/firebase_auth.dart';
 
 class ConfigListItem extends ConsumerWidget {
+  final String currentAuthor = FirebaseAuth.instance.currentUser!.uid;
   final String path;
+  String author="";
   final String entityId;
```
Comment (kovarny) [inline lib/entity/config/config_list_item.dart:14]: Instead of entityId let's pass through entityDocument which contains author that will not change.

**Review by kovarny — CHANGES_REQUESTED** [review]

File: lib/entity/config/config_list_item.dart (hunk 2 above)
Comment (kiennguyen2403) [reply lib/entity/config/config_list_item.dart:14]: I have checked the "selectedEntityDoc" in the config_list.dart and it is mostly null, or does not have the field "author". I don't know is it the entityDocument you are looking for?

File: lib/controls/doc_field_text_edit.dart [hunk 3]
```diff
@@ -22,11 +23,12 @@ class DocFieldTextEdit extends ConsumerStatefulWidget {
 
 class DocFieldTextEditState extends ConsumerState<DocFieldTextEdit> {
   Timer? descSaveTimer;
-
+  final dbInstance = FirebaseFirestore.instance;
```
Comment (kovarny) [inline lib/controls/doc_field_text_edit.dart:26]: not sure if this is used

**Review by kovarny — APPROVED** [review]

**Pull request merged**